server/src/main/java/com/thoughtworks/go/server/service/BuildAssignmentService.java

```diff
@@ -314,15 +315,15 @@ private Work createWork(final AgentInstance agent, final JobPlan job) {
         }
     }
 
-    private EnvironmentVariableContext getEnvironmentVariableContextFromEnvironment(String pipelineName) {
+    EnvironmentVariableContext getEnvironmentVariableContextFromEnvironment(String pipelineName) {
         EnvironmentConfig environmentConfig = environmentConfigService.environmentForPipeline(pipelineName);
         if (environmentConfig == null) {
             return null;
         }
 
         secretParamResolver.resolve(environmentConfig);
 
-        EnvironmentVariableContext environmentVariableContext = new EnvironmentVariableContext();
+        EnvironmentVariableContext environmentVariableContext = new EnvironmentVariableContext(GO_ENVIRONMENT_NAME, CaseInsensitiveString.str(environmentConfig.name()));
```

Comment:
Optional:

Can create context using `environmentConfig.createEnvironmentContext()`

Reply:
@GaneshSPatil and I spoke about this, calling `environmentConfig.createEnvironmentContext()` would mean loosing all the resolved environment variables. So let's not change it.